fern/api/definition/api/__package__.yml

```diff
@@ -0,0 +1,89 @@
+services: {}
+
+types:
+  ApiId: string
+
+  Api:
+    properties:
+      id: ApiId
+      displayName: string
+      definition: ApiDefinition
+
+  ApiDefinition:
+    properties:
+      services: list<Service>
+      types: map<TypeId, TypeDefinition>
+
+  Service:
+    properties:
+      endpoints: list<Endpoint>
+
+  Endpoint:
+    properties:
+      path: EndpointPath
+
+  EndpointPath:
+    properties:
+      parts: list<EndpointPathPart>
+
+  EndpointPathPart:
+    union:
+      literal: string
+      pathParameter: EndpointPathParameter
+
+  EndpointPathParameter:
+    properties:
+      name: string
+      type: TypeReference
+
+  TypeId: string
+
+  TypeDefinition:
+    union:
+      alias: TypeReference
+      enum: ...
+      union: ...
+      object: ObjectType
+
+  TypeReference:
+    union:
+      primitive: PrimitiveType
+      container: ContainerType
+      unknown: {}
+      reference: TypeId
+      definition: TypeDefinition
+
+  ObjectType:
+    properties:
+
+  ObjectPropertes:
+    properties:
+      key: string
+      value: TypeReference
+
+  PrimitiveType:
+    enum:
+      - integer
+      - double
+      - long
+      - string
+      - boolean
```

Comment:
can we make these unions -- i wanna make it easy to support `regex` in string for example ?